lib/matplotlib/backend_bases.py

```diff
@@ -3642,6 +3655,74 @@ def trigger_tool(self, name):
         """
         self.navigation.tool_trigger_event(name, sender=self)
 
+    def get_group_count(self, name):
+        """Gets useful indicies for a group.
+
+        Parameters
+        ----------
+        name : string
+            The name of the group.
+
+        Returns
+        -------
+        (start_index, group_pos) : where:
+          - `start_index, the index at the first item in the toolbar.
+          - `group_pos`, the index of this group within the toolbar.
+        """
+        index = 0
+        group_count = 0
+        for group in self._groups:
+            if group == name:
+                break
+            index += len(self._groups[group])
+            group_count += 1
+        return index, group_count
+
+    def get_positions(self, name, group, position=-1):
+        """Adds the tool to the group
+
+        Parameters
+        ----------
+        name : string
+            The name of the tool.
+        group : string
+            The name of the group.
+        position : int
+            The position within the group, if -1, it goes at the end.
+
+        Returns
+        -------
+        (position, insert_index) : where:
+
+          - `position` a validated version of the input argument, i.e. > 0.
+          - `insert index` the position used to insert into the toolbar.
+        """
+
+        if position < 0:
+            position += len(self._groups.get(group, [])) + 1
+            if position < 0:
+                position = 0
+
+        return position, self.get_group_count(group)[0] + position
+
+    def add_tool_to_group(self, tool, group, position=-1):
```

Comment:
I did it like this so that one can either add a tool by overall position if the backend doesn't support a way to implement groups, thus the backend can add the tool in ``add_toolitem`` using absolute position, or by this method using group position.

I don't think a framework exists where this would not work for.  I imagine you refer to one that does do groups, but doesn't return a toolitem, in which case the backend could just save the raw information from ``add_toolitem``.

I guess I try and implement here a default for the most basic case, i.e. the no-groups situation, so that we don't have to duplicate this code.